File: ShareBudget/UserAPI.m
//
//  UserAPI.m
//  ShareBudget
//
//  Created by Denys Meloshyn on 12.01.17.
//  Copyright © 2017 Denys Meloshyn. All rights reserved.
//

#import "UserAPI.h"

#import "BaseAPI.h"
#import "AsynchronousURLConnection.h"

@implementation UserAPI

+ (NSURLSessionTask *)update:(NSObject *)user completionBlock:(APICompletionBlock)completionBlock
{
    NSURLComponents *components = [BaseAPI components];
    components.path = @"/user";
    
    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:components.URL];
    request.HTTPMethod = @"PUT";
    
    [request addValue:@"last name" forHTTPHeaderField:kLastName];
    [request addValue:@"1" forHTTPHeaderField:kPassword];
    [request addValue:@"ned1988@gmail.com" forHTTPHeaderField:kEmail];
    [request addValue:@"name" forHTTPHeaderField:kFirstName];
    [request addValue:[@YES stringValue] forHTTPHeaderField:kIsRemoved];
    
    return [AsynchronousURLConnection createAsynchronousRequest:request completionBlock:completionBlock];
}

@end

File: ShareBudget/UserAPI.swift
//
//  UserAPI.swift
//  ShareBudget
//
//  Created by Denys Meloshyn on 13.01.17.
//  Copyright © 2017 Denys Meloshyn. All rights reserved.
//

import CoreData

class UserAPI: BaseAPI {
    override func timestampStorageKey() -> String {
        return "user_timestamp"
    }

    override func parseUpdates(items: [[String: Any?]], in managedObjectContext: NSManagedObjectContext) {
        var user: User?

        for item in items {
            if let modelID = item[User.modelKeyID()] as? Int {
                user = ModelManager.findEntity(User.self, by: modelID, in: managedObjectContext) as? User
            }

            if user == nil {
                user = User(context: managedObjectContext)
            }

            user?.update(with: item, in: managedObjectContext)
        }
    }
}
